File: server/src/models/UserSkill.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');

const UserSkill = sequelize.define('UserSkill', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  skillId: {
    type: DataTypes.INTEGER
  },
  skillName: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  type: {
    type: DataTypes.ENUM('teach', 'learn'),
    allowNull: false
  },
  experienceLevel: {
    type: DataTypes.ENUM('Beginner', 'Intermediate', 'Expert'),
    defaultValue: 'Beginner'
  },
  isVerified: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  },
  validationScore: {
    type: DataTypes.INTEGER,
    defaultValue: 0
  },
  certificateUrl: {
    type: DataTypes.STRING(500)
  },
  certificateVerified: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  }
}, {
  tableName: 'user_skills',
  timestamps: true
});

module.exports = UserSkill;

File: server/src/main/java/com/skillswap/entity/UserSkill.java
package com.skillswap.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "user_skills")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserSkill {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    private Long skillId;

    @Column(nullable = false, length = 100)
    private String skillName;

    @Column(nullable = false, length = 10)
    private String type; // "teach" or "learn"

    @Column(length = 20)
    @Builder.Default
    private String experienceLevel = "Beginner";

    @Builder.Default
    private Boolean isVerified = false;

    @Builder.Default
    private Integer validationScore = 0;

    @Column(length = 500)
    private String certificateUrl;

    @Builder.Default
    private Boolean certificateVerified = false;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
